streampipes-client-go/streampipes/internal/util/streampipes_api_path.go

```diff
@@ -0,0 +1,66 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package util
+
+import (
+	"strings"
+)
+
+type StreamPipesApiPath struct {
+	PathItems []string // PathItems stores URL fragments
+}
+
+func NewStreamPipesApiPath(initialPathItems []string) *StreamPipesApiPath {
+
+	return &StreamPipesApiPath{
+		PathItems: initialPathItems,
+	}
+}
+
+func (s *StreamPipesApiPath) FromStreamPipesBasePath() *StreamPipesApiPath {
+	path := "streampipes-backend"
+	s.PathItems = append(s.PathItems, path)
+	return s
+}
+
+func (s *StreamPipesApiPath) FromStreamPipesBasePathWithSubPaths(allSubPaths []string) *StreamPipesApiPath {
+	return s.FromStreamPipesBasePath().AddToPath(allSubPaths)
+}
+
+func (s *StreamPipesApiPath) AddToPath(pathItem []string) *StreamPipesApiPath {
+	s.PathItems = append(s.PathItems, pathItem...)
+	return s
+}
+
+func (s *StreamPipesApiPath) ToString() string {
```

Comment:
In golang, we use `String`

Reply:
"String" may overlap with the names of methods or functions in other libraries, but I think it can be changed to "UrlString"

Reply:
That's on purpose. We're implementing the Stringer interface. https://go.dev/tour/methods/17